### PR TITLE
Implement scheduled and recurring tests

### PR DESCRIPTION
**Pull request by samnissen**

### CODE REVIEW

File: app/views/scheduled_tests/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+<p>
+  <strong>Next test:</strong>
+  <%= @scheduled_test.next_test %>
+</p>
+
+<p>
+  <strong>Recurring:</strong>
+  <%= @scheduled_test.recurring %>
+</p>
```
Comment (jhulme) [inline app/views/scheduled_tests/show.html.erb:26]: Need to indicate the units of recurrence. Assuming this is days currently.

Also need to apply some level of validation as it is possible to provide a negative value here.

File: app/views/scheduled_tests/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      <tr class="scheduled-tests-index-row">
+        <td class="scheduled-test-last-test-column cursor-hand"
+            onclick="location.href='<%= edit_scheduled_test_path(scheduled_test) %>'">
+            <% run_status = scheduled_test.last_run_status %>
+          <div class="last-test-status last-test-<%= run_status[:class] %>"><%= run_status[:display] %></div>
+        </td>
+        <td class="scheduled-test-tests-column">
+          <div class="scheduled-test-testsets-container">
+          <% scheduled_test.test_ids.each do |test_id| %>
+            <a href="" class="btn btn-default scheduled-test-testset">
```
Comment (jhulme) [inline app/views/scheduled_tests/index.html.erb]: Clicking the testset buttons doesn't provide any response. Unsure whether this was meant to link through to the test set page or if there was never meant to be any functionality (in which case don't need to format as a button).

File: app/views/scheduled_tests/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+      <%= f.collection_select(:environment_id, Environment.where(user: current_user).order('name ASC'), :id, :name,
+      {},
+      { :class => 'select-list form-control environment-select-list' }) %>
+    </div>
+  </div>
+  <div class="form-group">
+    <%= f.label :next_test, :class => 'control-label col-md-2' %>
+    <div class="col-md-10">
+      <%= f.text_field :next_test, :class => 'form-control datepicker',
+                :id => "nextTestScheduledTest" %>
```
Comment (jhulme) [inline app/views/scheduled_tests/_form.html.erb:78]: The time here displays with the offset applied. In comparison the time displayed on the schedule tests index page does not. From a users perspective I would prefer all of the times displayed to me throughout the UI have my local time offset applied.

Comment (samnissen) [inline app/views/scheduled_tests/_form.html.erb:78]: The scheduled tests index does now use the users' timezone.

**Pull request merged**